File: lab-6.c
#include <stdio.h>

// Function to calculate the sum of an array
int calculate_sum(int arr[], int size) {
    int sum = 0, i = 0;
    do {
        sum += arr[i];
        i++;
    } while (i < size);
    return sum;
}

// Function to calculate the average of an array
float calculate_average(int sum, int size) {
    return (float)sum / size;
}

// Main function to take input and calculate sum and average
int main() {
    int n;

    // Input number of elements in the array
    do {
        printf("Enter the number of elements in the array (must be at least 1): ");
        scanf("%d", &n);
    } while (n <= 0);  // Ensures user enters a positive number

    int arr[n];

    // Input array elements
    for (int i = 0; i < n; i++) {
        printf("Enter element %d: ", i + 1);
        scanf("%d", &arr[i]);
    }

    // Calculate sum using the user-defined function
    int sum = calculate_sum(arr, n);
    printf("Sum of the array: %d\n", sum);

    // Calculate average using the user-defined function
    float average = calculate_average(sum, n);
    printf("Average of the array: %.2f\n", average);

    return 0;
}

File: lab-6.py

# Function to calculate the sum of an array
def calculate_sum(arr):
    total_sum = 0
    i = 0
    while True:
        total_sum += arr[i]
        i += 1
        if i >= len(arr):
            break
    return total_sum

# Function to calculate the average of an array
def calculate_average(arr, total_sum):
    return total_sum / len(arr)

# Main program to take input and calculate sum and average
def main():
    arr = []
    
    while True:
        n = int(input("Enter the number of elements in the array (must be at least 1): "))
        if n > 0:
            break
        print("Please enter a valid number greater than 0.")
    
    for i in range(n):
        element = float(input(f"Enter element {i+1}: "))
        arr.append(element)

    # Calculate sum using the defined function
    total_sum = calculate_sum(arr)
    print(f"Sum of the array: {total_sum}")

    # Calculate average using the defined function
    average = calculate_average(arr, total_sum)
    print(f"Average of the array: {average}")

# Run the main program
main()
